browser/src/core/Socket.js

```diff
@@ -31,7 +31,6 @@ app.definitions.Socket = L.Class.extend({
 		this._map = map;
 		this._msgQueue = [];
 		this._delayedMessages = [];
-		this._handlingDelayedMessages = false;
```

Comment:
I assume this is a re-entrancy guard to ensure that if more messages arrive while we are processing others (not sure if/how that can happen), that they are appended to the delay queue rather than being processed out of order.

Reply:
yeah, but the question is if that is at all possible?

Reply:
Fair cop - if you think not; lets simplify it; but mis-ordering of events in races/corner-cases adds -such- difficult to find/fix/reproduce bugs - that I'd love some vestigal warning / assert if its possible =) some defensiveness may be useful.